src/domains/chart/components/lib-charts/sparkline-chart.tsx

```diff
@@ -30,7 +30,14 @@ const getForceTimeWindowCorrection = (
 
   const currentDuration = before - after
   const requestedDuration = requestedBefore - requestedAfter
-  const widthRatio = (currentDuration / requestedDuration)
+  // don't do overrides when current (available) duration is bigger or only slightly lower
+  // than requested duration
+  const DURATION_CHANGE_TOLERANCE = 1.03
+  if (currentDuration > requestedDuration / DURATION_CHANGE_TOLERANCE) {
```

Comment:
and requestDuration never ever could be zero?

Reply:
no, never, it could mean setting `after=0` and `before=0`